### PR TITLE
✅ Compare View Tests and Karma removal

### PR DESCRIPTION
**Pull request by dominikkaminski**
**Changes:**

1. Remove Karma (Old and unused test framework | Protractor replaces Karma)
2. Add Compare view test

PR: #858 
---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: aurelia_project/tasks/e2e.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ function clean(): Promise<Array<string>> {
 }
 
 function build(): NodeJS.ReadWriteStream {
-  delete tsConfig.compilerOptions.lib;
+  // delete tsConfig.compilerOptions.lib;
```
Comment (robpal94) [inline aurelia_project/tasks/e2e.ts]: Remove this out-commented line or add a todo comment, why this was out-commented.

File: src/modules/navbar/navbar.ts [hunk 2]
```diff
@@ -88,6 +89,20 @@ export class NavBar {
...
+
+    this._eventAggregator.subscribe(environment.events.diagramChange, () => {
+      this.containsUnsavedChanges = true;
+    });
+
+    this._eventAggregator.subscribe(environment.events.processDefDetail.saveDiagram, () => {
+      this.containsUnsavedChanges = false;
+    });
+
+>>>>>>> :sparkles: Add Edited Label When Diagram is Changed
```
Comment (robpal94) [inline src/modules/navbar/navbar.ts]: Merge conflict marker

File: test/e2e/src/bpmn-xml-view.spec.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      return general.getRouterViewContainer;
+    });
+
+    // You have to open solution explorer before click on link
+    navBar.openSolutionExplorerByButtonClick();
+    solutionExplorer.openProcessModelByClick(processModelId);
+
+    // Wait until diagram is loaded
+    browser.driver.wait(() => {
+      browser.wait(expectedConditions.visibilityOf(bpmnIo.bpmnIoTag), defaultTimeoutMS);
```
Comment (robpal94) [inline test/e2e/src/bpmn-xml-view.spec.ts]: We should offload the first argument into a constant. 

```
const bpmnIoTagVisible: boolean = expectedConditions.visibilityOf(bpmnIo.bpmnIoTag);
browser.wait(bpmnIoTagVisible, defaultTimeoutMS);
```

File: test/e2e/src/bpmnDiffView.spec.ts [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    processModelId = processModel.getProcessModelId();
+
+    // Create a new process definition by POST REST call
+    processModel.postProcessModelWithUserTask(processModelId);
+  });
+
+  beforeEach(() => {
+    browser.get(aureliaUrl);
+    browser.driver.wait(() => {
+      browser.wait(expectedConditions.visibilityOf(general.getRouterViewContainer), defaultTimeoutMS);
```
Comment (robpal94) [inline test/e2e/src/bpmnDiffView.spec.ts]: See above

File: test/e2e/src/bpmnDiffView.spec.ts [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+      return general.getRouterViewContainer;
+    });
+
+    // You have to open solution explorer before click on link
+    navBar.openSolutionExplorerByButtonClick();
+    solutionExplorer.openProcessModelByClick(processModelId);
+
+    // Wait until diagram is loaded
+    browser.driver.wait(() => {
+      browser.wait(expectedConditions.visibilityOf(bpmnIo.bpmnIoTag), defaultTimeoutMS);
```
Comment (robpal94) [inline test/e2e/src/bpmnDiffView.spec.ts]: See above

File: test/e2e/src/dashboard.spec.ts [hunk 6]
```diff
@@ -14,6 +14,7 @@ describe('Dashboard view', () => {
 
   const aureliaUrl: string = browser.params.aureliaUrl;
   const defaultTimeoutMS: number = browser.params.defaultTimeoutMS;
+  const taskListTimeoutMS: number = defaultTimeoutMS + defaultTimeoutMS;
```
Comment (robpal94) [inline test/e2e/src/dashboard.spec.ts]:
```
const taskListTimeoutMS: number = 2 * defaultTimeoutMS;
```

File: test/e2e/src/dashboard.spec.ts [hunk 7]
```diff
@@ -40,4 +42,145 @@ describe('Dashboard view', () => {
...
+
+  it('should contain process list.', () => {
+    dashboard.processListTag.isDisplayed().then((dashboardProcessListTagIsDisplayed: boolean) => {
+      expect(dashboardProcessListTagIsDisplayed).toBeTruthy();
+    });
+  });
+
+  it('should contain at least process definition in running section.', () => {
+    browser.driver.wait(() => {
+      browser.wait(expectedConditions.visibilityOf(dashboard.firstProcessRunningListItems), defaultTimeoutMS);
```
Comment (robpal94) [inline test/e2e/src/dashboard.spec.ts]: See above

File: test/e2e/src/dashboard.spec.ts [hunk 8]
```diff
@@ -40,4 +42,145 @@ describe('Dashboard view', () => {
...
+    });
+
+    dashboard.countOfProcessRunningListItems().then((countOfProcessRunningListItems: number) => {
+      expect(countOfProcessRunningListItems).toBeGreaterThanOrEqual(1);
+    });
+  });
+
+  it('should contain recently started process in running section.', () => {
+    browser.driver.wait(() => {
+      browser.wait(expectedConditions.visibilityOf(dashboard.firstProcessRunningListItemsById(processModel.getCorrelationId())), defaultTimeoutMS);
```
Comment (robpal94) [inline test/e2e/src/dashboard.spec.ts]: `please(avoid(chained(method(calls()))))`

File: test/e2e/src/dashboard.spec.ts [hunk 9]
```diff
@@ -40,4 +42,145 @@ describe('Dashboard view', () => {
...
+
+    dashboard.countOfProcessRunningListItemsByCorrelationId(processModel.getCorrelationId()).then((countOfProcessRunningListItems: number) => {
+      expect(countOfProcessRunningListItems).toBe(1);
+    });
+  });
+
+  it('should be possible to open process model by click on hyperlink in table.', () => {
+    browser.driver.wait(() => {
+      browser.wait(expectedConditions.visibilityOf(
+        dashboard.hyperlinkOfProcessRunningListItemByCorrelationId(processModel.getCorrelationId())), defaultTimeoutMS);
```
Comment (robpal94) [inline test/e2e/src/dashboard.spec.ts]: see above

File: test/e2e/src/dashboard.spec.ts [hunk 10]
```diff
@@ -40,4 +42,145 @@ describe('Dashboard view', () => {
...
+      browser.getCurrentUrl().then((currentBrowserUrl: string) => {
+        expect(currentBrowserUrl).toContain(processModel.processModelUrl(processModel.getProcessModelId()));
+      });
+    });
+  });
+
+  it('should be possible to open user tasks by click on hyperlink in table.', () => {
+    browser.driver.wait(() => {
+      browser.wait(expectedConditions.visibilityOf(
+        dashboard.hyperlinkOfUserTasksInProcessRunningListItemByCorrelationId(processModel.getCorrelationId())), defaultTimeoutMS);
```
Comment (robpal94) [inline test/e2e/src/dashboard.spec.ts]: See above

File: test/e2e/src/dashboard.spec.ts [hunk 11]
```diff
@@ -40,4 +42,145 @@ describe('Dashboard view', () => {
...
+    });
+
+    dashboard.countOfTasksWaitingListItems().then((countOfTasksWaitingListItems: number) => {
+      expect(countOfTasksWaitingListItems).toBeGreaterThanOrEqual(1);
+    });
+  });
+
+  it('should contain recently started task in tasks waiting section.', () => {
+    browser.driver.wait(() => {
+      browser.wait(expectedConditions.visibilityOf(dashboard.firstTaskWaitingById(processModelId)), taskListTimeoutMS);
```
Comment (robpal94) [inline test/e2e/src/dashboard.spec.ts]: chained method calls

File: test/e2e/src/dashboard.spec.ts [hunk 12]
```diff
@@ -40,4 +42,145 @@ describe('Dashboard view', () => {
...
+    });
+
+    dashboard.countOfTasksWaitingListItemsById(processModelId).then((countOfTasksWaitingListItemsById: number) => {
+      expect(countOfTasksWaitingListItemsById).toBe(1);
+    });
+  });
+
+  it('should be possbible to click continue in task waiting section.', () => {
+    browser.driver.wait(() => {
+      browser.wait(expectedConditions.visibilityOf(dashboard.firstTaskWaitingById(processModelId)), taskListTimeoutMS);
```
Comment (robpal94) [inline test/e2e/src/dashboard.spec.ts]: chained method calls

File: test/e2e/src/dashboard.spec.ts [hunk 13]
```diff
@@ -40,4 +42,145 @@ describe('Dashboard view', () => {
...
+    dashboard.continueTaskByClickOnButton(processModelId).then(() => {
+      browser.getCurrentUrl().then((currentBrowserUrl: string) => {
+        expect(currentBrowserUrl).toContain(processModel.userTasksInputUrl(processModel.getProcessModelId()));
+      });
+    });
+  });
+
+  it('should be possbible to click continue in an opened user task.', () => {
+    browser.driver.wait(() => {
+      browser.wait(expectedConditions.visibilityOf(dashboard.firstTaskWaitingById(processModelId)), taskListTimeoutMS);
```
Comment (robpal94) [inline test/e2e/src/dashboard.spec.ts]: chained method calls

File: test/e2e/src/dashboard.spec.ts [hunk 14]
```diff
@@ -40,4 +42,145 @@ describe('Dashboard view', () => {
...
+    // Wait until view is loaded and button is visible
+    browser.driver.wait(() => {
+      browser.wait(expectedConditions.visibilityOf(dashboard.dynamicUiWrapperContinueButton), taskListTimeoutMS);
+      return dashboard.dynamicUiWrapperContinueButton;
+    });
+  });
+
+  it('should be in waiting room when click continue in an opened user task.', () => {
+    browser.driver.wait(() => {
+      browser.wait(expectedConditions.visibilityOf(dashboard.firstTaskWaitingById(processModelId)), taskListTimeoutMS);
```
Comment (robpal94) [inline test/e2e/src/dashboard.spec.ts]: chained method calls

File: test/e2e/src/dashboard.spec.ts [hunk 15]
```diff
@@ -40,4 +42,145 @@ describe('Dashboard view', () => {
...
+      return dashboard.firstTaskWaitingById(processModelId);
+    });
+
+    dashboard.continueTaskByClickOnButton(processModelId);
+
+    // Wait until view is loaded and button is visible
+    browser.driver.wait(() => {
+      browser.wait(expectedConditions.visibilityOf(dashboard.dynamicUiWrapperContinueButton), taskListTimeoutMS);
```
Comment (robpal94) [inline test/e2e/src/dashboard.spec.ts]: chained method calls

File: test/e2e/src/dashboard.spec.ts [hunk 16]
```diff
@@ -40,4 +42,145 @@ describe('Dashboard view', () => {
...
+      return dashboard.dynamicUiWrapperContinueButton;
+    });
+
+    // Should be in waiting room
+    dashboard.continueUserTaskByClickOnDynamicUiWrapperContinuButton().then(() => {
+      browser.getCurrentUrl().then((currentBrowserUrl: string) => {
+        expect(currentBrowserUrl).toContain(processModel.waitingroomUrl(processModelId, processModel.getCorrelationId()));
```
Comment (robpal94) [inline test/e2e/src/dashboard.spec.ts]: chained method calls

File: test/e2e/src/dashboard.spec.ts [hunk 17]
```diff
@@ -40,4 +42,145 @@ describe('Dashboard view', () => {
...
+      });
+    });
+
+    // Should be in dashboard view
+    browser.driver.wait(() => {
+      browser.wait(expectedConditions.urlContains(dashboard.dashboardLink), taskListTimeoutMS);
```
Comment (robpal94) [inline test/e2e/src/dashboard.spec.ts]: chained method calls

File: test/e2e/src/pages/bpmn-io.ts [hunk 18]
```diff
@@ -0,0 +1,7 @@
+import {by, element, ElementFinder} from 'protractor';
+
+export class BpmnIo {
+
+  // Define Elements
+  public bpmnIoTag: ElementFinder = element(by.tagName('bpmn-io'));
```
Comment (robpal94) [inline test/e2e/src/pages/bpmn-io.ts]: chained method calls

File: test/e2e/src/pages/bpmn-xml-view.ts [hunk 19]
```diff
@@ -0,0 +1,16 @@
+import {by, element, ElementFinder} from 'protractor';
+
+export class BpmnXmlView {
+
+  // Define Links, Urls, Classes
+  public bpmnLineNumbersClass: string = 'hljs-ln-numbers';
+
+  // Define Elements
+  public bpmnXmlViewTag: ElementFinder = element(by.tagName('bpmn-xml-view'));
```
Comment (robpal94) [inline test/e2e/src/pages/bpmn-xml-view.ts]: chained method calls

File: test/e2e/src/pages/bpmnDiffView.ts [hunk 20]
```diff
@@ -0,0 +1,18 @@
+import {by, element, ElementFinder} from 'protractor';
+
+export class BpmnDiffView {
+
+  // Define Links, Urls, Classes
+
+  // Define Elements
+  public bpmnDiffViewTag: ElementFinder = element(by.tagName('bpmn-diff-view'));
+  public changesListClassName: ElementFinder = element(by.className('diff-change-list'));
```
Comment (robpal94) [inline test/e2e/src/pages/bpmnDiffView.ts]: chained method calls

File: test/e2e/src/pages/dashboard.ts [hunk 21]
```diff
@@ -1,6 +1,71 @@
...
+
+  public async countOfTasksWaitingListItems(): Promise<number> {
+    return this.taskListContainer.all(by.className(this._domTaskClassName)).count();
+  }
+  public async countOfTasksWaitingListItemsById(processModelId: string): Promise<number> {
+    return this.taskListContainer.all(by.id(this._domTaskIdPrefix + processModelId)).count();
+  }
+
+  public continueTaskByClickOnButton(processModelId: string): any {
+    return this.firstTaskWaitingById(processModelId).element(by.className('task-list-continue-button')).click();
```
Comment (robpal94) [inline test/e2e/src/pages/dashboard.ts]: Chained method call (also above)

File: test/e2e/src/pages/processModel.ts [hunk 22]
```diff
@@ -7,21 +7,43 @@ export class ProcessModel {
...
+  public userTasksUrl(processModel: string): string {
+    return '/processdef/' + processModel + '/task';
+  }
+
+  public userTasksInputUrl(processModel: string): string {
+    return '/processdef/' + processModel + '/task/Task_0k8m2if/dynamic-ui';
+  }
+
+  public waitingroomUrl(processModelId: string, correlationId: string): string {
+    return '/waitingroom/' + correlationId + '/' + processModelId;
```
Comment (robpal94) [inline test/e2e/src/pages/processModel.ts]: Please use template strings here.

File: test/e2e/src/bpmn-xml-view.spec.ts (hunk 3 above)
Comment (SteffenKn) [reply test/e2e/src/bpmn-xml-view.spec.ts]: With spacing:
```
const bpmnIoTagVisible: boolean = expectedConditions.visibilityOf(bpmnIo.bpmnIoTag);

browser.wait(bpmnIoTagVisible, defaultTimeoutMS);
```

**Pull request merged**